br.unb.cic.mcsl/src/br/unb/cic/mcsl/generator/MetaCrySLGenerator.xtend

```diff
@@ -35,22 +55,78 @@ class MetaCrySLGenerator extends AbstractGenerator {
 //				.join(', '))
 	}
 	
+	def Refinement mergeRefinements(String name, ArrayList<Refinement> refs) {
+		// TODO: receives a list of refinements modules and returns one module with all merged
+		val newRefinement = (new MetaCrySLFactoryImpl()).createRefinement()
+		val operationsList = new ArrayList<RefinementOpr>;
+		newRefinement.name = name
+		
+		// Creates a flat list with all refinement operations
+		val oprList = refs.map(r | r.refinements)
+		for(list: oprList) {
+			for(el: list) {
+				operationsList.add(el)
+			}
+		}
+		
+		for(el: operationsList) {
+			println(el)
+			newRefinement.refinements.add(el)
+		}
```

Comment:
Hi @vinicius0197 . I think that this design works for most of the cases. Nonetheless, merging the "define variable" operations is a bit different. Consider the following case: 

   * ref 1 sets algorithm to {"AES", "BLOWFISH"} 
   * ref 2 sets algorithm to {"TripleDES"}

If we merge ref 1 and ref 2, algorithm must be set to {"AES", "BLOWFISH", "TripleDES"}. 

Perhaps we should have a meeting to discuss this issue.

Reply:
That's true @rbonifacio . Maybe adding a check just for this case would do the trick? I'm thinking about the `rename` operation as well.

Surely we can have a meeting to discuss this. Maybe Monday morning? In the meantime, I'm trying to set up the code for writing the final CrySL files to disk (even if some of the merges are still wrong, I thought it would be valuable to have the complete workflow running from the start)